### PR TITLE
Update AAD & AAD B2C docs

### PR DESCRIPTION
**Pull request by moarychan**
Changelog:
- Update AAD & B2C migration doc.
- Update Security doc.

### CODE REVIEW

**Comment by moarychan** [comment]
https://github.com/Azure/azure-sdk-for-java/pull/27700

**Comment by moarychan** [comment]
https://github.com/Azure-Samples/azure-spring-boot-samples/pull/223

**Comment by moarychan** [comment]
https://github.com/Azure/azure-sdk-for-java/pull/27727

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 1]
```diff
@@ -301,49 +301,75 @@ crash.
 [cols="<30,<~", options="header"]
 |===
 |Removed dependencies |Description
+|com.fasterxml.jackson.core:jackson-databind |Please include the jackson databind dependency if you want to use Jackson Databind.
+|io.projectreactor.netty:reactor-netty |Please include the reactor netty dependency if you want to use Reactor Netty.
 |org.springframework.boot:spring-boot-starter-validation |Please include the validation starter if you want to use Hibernate Validator.
+|org.springframework.boot:spring-boot-starter-webflux |Please include the webflux starter if you want to use AAD condition access.
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:307]: Please include the webflux starter if you want to use Web Flux.

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 2]
```diff
@@ -301,49 +301,75 @@ crash.
...
 |===
 
 [#configuration-spring-cloud-azure-starter-active-directory]
 ===== SDK Configuration Changes
 
 . All configuration property names changed the prefix from `azure.activedirectory` to `spring.cloud.azure.active-directory`.
-. New property `spring.cloud.azure.active-directory.enabled=true` is necessary to enable related features.
+. New property `spring.cloud.azure.active-directory.enabled=true` is required to enable related features.
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:314]: New property `spring.cloud.azure.active-directory.enabled=true` is added to allow enable / disable AAD related features. The default vaule is `true`.

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 3]
```diff
@@ -301,49 +301,75 @@ crash.
...
+| *azure.activedirectory*.user-group.allowed-group-names                                     | *spring.cloud.azure.active-directory*.user-group.allowed-group-names
+| *azure.activedirectory*.user-name-attribute                                                | *spring.cloud.azure.active-directory*.user-name-attribute
+|===
+
+[NOTE]
+====
+The value type of the following properties is changed from `long` to `java.time.Duration`, they are `jwt-connect-timeout`, `jwt-read-timeout`, `jwk-set-cache-lifespan`, `jwk-set-cache-refresh-time`.
+====
+
+.Removed property from azure-spring-boot-starter-active-directory
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:354]: properties

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 4]
```diff
@@ -354,12 +380,58 @@ NOTE: `azure.activedirectory.graph-membership-uri` has been replaced by 2 proper
...
+|com.azure.spring.autoconfigure.aad.Membership |com.azure.spring.cloud.autoconfigure.aad.implementation.graph.Membership
+|com.azure.spring.autoconfigure.aad.Memberships |com.azure.spring.cloud.autoconfigure.aad.implementation.graph.Memberships
+|com.azure.spring.autoconfigure.aad.UserPrincipal |com.azure.spring.cloud.autoconfigure.aad.filter.UserPrincipal
+|com.azure.spring.autoconfigure.aad.UserPrincipalManager |com.azure.spring.cloud.autoconfigure.aad.filter.UserPrincipalManager
+|com.azure.spring.autoconfigure.condition.aad.ClientRegistrationCondition |com.azure.spring.cloud.autoconfigure.aad.implementation.conditions.ClientRegistrationCondition
+|com.azure.spring.autoconfigure.condition.aad.ResourceServerCondition |com.azure.spring.cloud.autoconfigure.aad.implementation.conditions.ResourceServerCondition
+|com.azure.spring.autoconfigure.condition.aad.WebApplicationCondition |com.azure.spring.cloud.autoconfigure.aad.implementation.conditions.WebApplicationCondition
+|===
+
+.Removed class from azure-spring-boot-starter-active-directory
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:428]: classes

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc [hunk 5]
```diff
@@ -391,25 +463,45 @@ crash.
 ===== SDK Configuration Changes
 
 . All configuration property names changed the prefix from `azure.activedirectory.b2c` to `spring.cloud.azure.active-directory.b2c`.
-. New property `spring.cloud.azure.active-directory.b2c.enabled=true` is necessary to enable related features.
+. New property `spring.cloud.azure.active-directory.b2c.enabled=true` is required to enable related features.
```
Comment (saragluna) [inline docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:466]: same for this one

File: docs/src/main/asciidoc/_migration-guide-for-4.0.adoc (hunk 2 above)
Comment (moarychan) [reply docs/src/main/asciidoc/_migration-guide-for-4.0.adoc:314]: OK, the default value is false.

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**